analyses/infercnv-consensus-cell-type/exploratory-notebooks/reference-query-counts.Rmd

```diff
@@ -0,0 +1,313 @@
+
+
+In total, this fraction of ScPCA samples have sufficient reference cells to run through `inferCNV`:
+
+```{r}
+sum(sample_counts_df$reference >= params$n_reference_threshold) / total_samples
+```
+
+Of the samples with sufficient reference cells, do any have _fewer_ query than reference cells?
+
+```{r}
+sample_counts_df |>
+  # consider usable only
+  dplyr::filter(reference >= params$n_reference_threshold) |>
+  dplyr::filter(query < params$n_reference_threshold)
+```
+
+There is one library with about 7x as many reference as query cells; let's look more closely:
+
+```{r}
+consensus_df |>
+  dplyr::filter(scpca_library_id == "SCPCL000021") |>
+  dplyr::count(consensus_annotation) |>
+  dplyr::arrange(n)
+```
+
+This sample indeed is dominated by immune cell types with very low tumor purity, but it still seems to have enough query cells to not be a problem and should be ok to run.
+
+## Plots
+
+In this section, we plot the relative contributions of each cell type group in the samples we can run. 
+We show the relative proportions of consensus cell type groups in the _reference_, and we show all other query cells grouped together in gray.
+The x-axis is library ids, and the y-axis is the proportion of each cell type group. 
+
+```{r}
+usable_libraries <- sample_counts_df |>
+  dplyr::filter(reference >= params$n_reference_threshold) |>
+  dplyr::pull(scpca_library_id)
+
+plot_df <- query_reference_df |>
+  dplyr::filter(scpca_library_id %in% usable_libraries) |>
+  dplyr::left_join(validation_group_df, by = c("consensus_ontology", "consensus_annotation")) |>
+  dplyr::select(
+    scpca_project_id, 
+    scpca_library_id, 
+    barcodes, 
+    cell_class, 
+    validation_group_annotation
+  ) |>
+  dplyr::mutate(
+    label = ifelse(
+      cell_class == "query", 
+      "query",
+      validation_group_annotation
+    ),
+    # move "query" first
+    label = forcats::fct_relevel(label, "query", after = 0)
+  )
+```
+
+
+```{r fig.width = 16, fig.height = 14}
+ggplot(plot_df) + 
+  aes(x = scpca_library_id, fill = label) + 
+  geom_bar(position = "fill") +
+  scale_fill_manual(values = celltype_pal) + 
+  facet_wrap(
+    vars(scpca_project_id), 
+    scales = "free_x", 
+    ncol = 4
+  ) + 
+  ylab("proportion") +
+  theme(
+    axis.text.x = element_blank(), 
+    axis.text.y = element_text(size = 12),
+    axis.title = element_text(size = 14),
+    strip.text = element_text(size = 13),
+    legend.text = element_text(size = 12),
+    legend.position = "bottom"
+  )
+```
+
+
+## Conclusions
+
+Currently, _before_ incorporating `SCimilarity` into our consensus cell types, we are able to run about 44% of samples through `inferCNV`. 
+Most broad diagnosis groups are represented, except `Mixed phenotype acute leukemia` and a handful of other solid tumors.
+All eligible projects except `SCPCP000011` (Retinoblastoma samples) are represented as well.
+
```

Comment:
Can you just add a sentence here that SCPCP000009 contains multiplexed libraries which will have pooled references. Just so we have it written somewhere.